service/awsconfig/v7/resource/lifecycle/current.go

```diff
@@ -2,9 +2,106 @@ package lifecycle
+			r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster cloud formation stack output values are not accessible due to stack state transition")
+			return nil, nil
+
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "found the guest cluster cloud formation stack in the AWS API")
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the guest cluster worker ASG name in the cloud formation stack")
+
+	var workerASGName string
+	{
+		workerASGName, err = r.service.GetOutputValue(stackOutputs, key.WorkerASGKey)
+		if cloudformationservice.IsOutputNotFound(err) {
+			// Since we are transitioning between versions we will have situations in
+			// which old clusters are updated to new versions and miss the ASG name in
+			// the CF stack outputs. We stop resource reconciliation at this point to
+			// reconcile again at a later point when the stack got upgraded and
+			// contains the ASG name.
+			//
+			// TODO remove this condition as soon as all guest clusters in existence
+			// obtain a ASG name.
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster worker ASG name in the cloud formation stack")
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource reconciliation for custom object")
+			return nil, nil
+
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "found the guest cluster worker ASG name in the cloud formation stack")
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("looking for the guest cluster instances being in state '%s'", autoscaling.LifecycleStateTerminatingWait))
+
+	var instances []*autoscaling.Instance
+	{
+		i := &autoscaling.DescribeAutoScalingGroupsInput{
+			AutoScalingGroupNames: []*string{
+				aws.String(workerASGName),
+			},
+		}
+
+		o, err := r.clients.AutoScaling.DescribeAutoScalingGroups(i)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		for _, g := range o.AutoScalingGroups {
+			for _, i := range g.Instances {
+				if *i.LifecycleState == autoscaling.LifecycleStateTerminatingWait {
+					instances = append(instances, i)
+				}
+			}
+		}
+
+		if len(instances) == 0 {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find the guest cluster instances being in state '%s'", autoscaling.LifecycleStateTerminatingWait))
+			return nil, nil
+
+		} else {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d guest cluster instances being in state '%s'", len(instances), autoscaling.LifecycleStateTerminatingWait))
+		}
+	}
+
+	// TODO check if NodeConfig exists
+	// TODO create NodeConfig if it does not exist
+	// TODO check if NodeConfig has final state
+	// TODO complete lifecycle hook if NodeConfig has final state
```

Comment:
I am thinking to derive another PR from this feature branch to split things up. Otherwise this here gets a bit too big. Testing would be easier when this stays smaller.

Reply:
The current size is OK but if the PR needs to be bigger then I'd appreciate splitting.

Reply:
I noticed we can just complete the lifecycle hooks right away to get this PR merged. Then we have the lifecycle hooks working and can further integrate the draining based on master. Scaling is then also fine. I will test this.